adding 8 new actions and a trigger

KL0maboi opened the pull request:
adding 
`removeCustomButtonFromPlayer`
`hideCustomButtonFromPlayer`
`showCustomButtonForPlayer`
`toggleCustomButtonVisibilityForPlayer`
`disableCustomButtonForPlayer`
`enableCustomButtonForPlayer`
`toggleCustomButtonUsabilityForPlyer`
actions and a `when player clicks button` trigger

![image](https://user-images.githubusercontent.com/85732105/159156743-d31fc4d0-4ec8-4f99-8486-85355d209887.png)

KL0maboi committed:
Update PlayerUiComponent.js

adding a warning message before opening links

KL0maboi committed:
Update ActionComponent.js

KL0maboi committed:
Update ClientNetworkEvents.js

KL0maboi committed:
Update PlayerUiComponent.js

KL0maboi committed:
Update server.js

KL0maboi committed:
Update ServerNetworkEvents.js

KL0maboi committed:
Update PlayerUiComponent.js

KL0maboi commented:
![image](https://user-images.githubusercontent.com/85732105/159156723-5630a565-7737-4221-9b78-61956fd93d27.png)

m0dE commented on server/ServerNetworkEvents.js:
```diff
@@ -781,6 +781,16 @@ var ServerNetworkEvents = {
+	_onPlayerButtonClick: function (data, clientId) {
+		var player = ige.game.getPlayerByClientId(clientId);
+		if(player) {
+			ige.trigger.fire("playerClickButton", {
```
rename as playerClicksCustomButton

m0dE commented on src/gameClasses/ClientNetworkEvents.js:
```diff
@@ -258,7 +258,9 @@ var ClientNetworkEvents = {
 				ige.playerUi.showCustomModal(data);
 				break;
-
+			case 'addButton':
```
rename as addCustomButton

m0dE commented on src/gameClasses/components/script/ActionComponent.js:
```diff
@@ -2586,6 +2586,17 @@ var ActionComponent = IgeEntity.extend({
 						}
 						break;
+						
+					case 'addButtonGuiForPlayer':
```
addCustomButtonForPlayer

m0dE commented on src/gameClasses/components/ui/PlayerUiComponent.js:
```diff
@@ -443,7 +443,29 @@ var PlayerUiComponent = IgeEntity.extend({
 		}
-	}
+	},
+	addButton: function (data) {
```
addCustomButton

m0dE commented:
this is a very interesting PR. thanks!

KL0maboi committed:
Update ServerNetworkEvents.js

KL0maboi committed:
Update ServerNetworkEvents.js

KL0maboi committed:
Update server.js

KL0maboi committed:
Update ClientNetworkEvents.js

KL0maboi committed:
Update ActionComponent.js

KL0maboi committed:
Update PlayerUiComponent.js

KL0maboi committed:
Update ClientNetworkEvents.js

KL0maboi committed:
adding 7 actions to customize buttons

added
removeCustomButtonFromPlayer
hideCustomButtonFromPlayer
showCustomButtonForPlayer
toggleCustomButtonVisibilityForPlayer
disableCustomButtonForPlayer
enableCustomButtonForPlayer
toggleCustomButtonUsabilityForPlyer

KL0maboi committed:
fixing typos

just fixing typos i made

KL0maboi commented:
i've added another 7 actions to customize buttons and you can test them with the json of a game i made i've attached it below
[game.txt](https://github.com/moddio/taro/files/8396764/game.txt)
